Demo/DemoTests/MoyaProviderSpec.swift

```diff
@@ -226,6 +256,31 @@ class MoyaProviderSpec: QuickSpec {
                         expect(provider.inflightRequests.count).to(equal(0))
                     }
                 })
+                
+                describe("a reactive provider with delayed stubs") {
+                    var provider: ReactiveMoyaProvider<GitHub>!
+                    beforeEach {
+                        let closure = { (target: GitHub) -> (Moya.StubbedBehavior) in
+                            return .Delayed(seconds: 2)
+                        }
+                        
+                        provider = ReactiveMoyaProvider(endpointsClosure: endpointsClosure, stubResponses: true, stubBehavior: closure)
+                    }
+                    
+                    it("returns canceled error when delayed execution is canceled") {
+                        var receivedError: NSError?
+                        let target: GitHub = .Zen
+                        waitUntil(timeout: 3) { done in
+                            let disposable = provider.request(target).subscribeError { (error) -> Void in
+                                receivedError = error
+                                done()
+                            }
+                            disposable.dispose()
+                        }
+                        
+                        expect(receivedError).toNot(beNil())
```

Comment:
So, this is tricky to test. The disposable will invoke its block when the subscription is ended, which is :+1: However, when you cal `disposable.dispose()`, that means that the `subscribeError` subscription is now disposed and will never be invoked. 

This is really really hard – the request is cancelled after its signal is not subscribed to anymore, which is behaviour _on top_ of the regular `MoyaProvider`. In practical use, no one using the RAC extension wants to know if a request was cancelled; it has only been cancelled because they no longer care about it. 

I think what we should test is that after disposing of the subscription, the `waitUntil` times out and `done()` is never invoked. Thoughts?